### DISCUSSION TITLE
Get the positon of the BarCode when the BarCode content is unreadable

### DISCUSSION THREAD

**Discussion by mangohurricane** [zxing-cpp/zxing-cpp#732] [Q&A] [answered]
Could you please provide guidance on how to determine the position of a barcode even when its content is unreadable?

**Comment by axxel** [comment, marked as answer]
If you are interested in 2D, meaning matrix, barcodes, then you can set the `returnErrors` flag of `ReaderOptions` (alias `DecodeHints`) to true. This will result in a barcode object with empty content and some `error` information set but the `position` will be set as well.

This generally does not work for 1D / linear barcodes, though.

**Reply by mangohurricane** [reply]
This is very helpful, Thank you for your great work on this project
